handler.py

```diff
@@ -271,19 +276,35 @@ def main():
         overlay_files = [x for x in overlay_files]
         overlay_mosaic = create_mosaic(overlay_files, Path(f"{args.output_directory}/mosaics/overlay.tif"))
 
+    # Get files for creating shapefile and/or pushing to AGOL
+    if args.create_shapefile or args.agol_user:
+        dmg_files = get_files(Path(args.output_directory) / 'dmg')
+
     if args.create_shapefile:
         print('Creating shapefile')
-        files = get_files(Path(args.output_directory) / 'dmg')
-        create_shapefile(files,
+        create_shapefile(dmg_files,
                          Path(args.output_directory).joinpath('shapes') / 'damage.shp',
                          args.destination_crs)
 
+    if args.agol_user and args.agol_password and args.agol_feature_service and args.agol_layer_num:
```

Comment:
This needs an else with the proper error message if all these things aren't included. Ideally, this would be checked for at the start of the script. If any, but not all, of those flags are set, throw the error. Otherwise, continue.

Reply:
Now checked and sets the agol_push to determine if enough variables are passed to push the AOI and centroids.